### PR TITLE
Add a comment in Image.toByteData to limit more encoders

### PR DESCRIPTION
**Pull request by eggfly**
* https://github.com/flutter/flutter/issues/94026

And besides I found these issues and discussions are related to Image.toByteData:
* https://github.com/flutter/flutter/issues/66299
* https://github.com/flutter/flutter/issues/16635
* https://github.com/flutter/flutter/issues/16537

I left documents and comments in the `toByteData()` API, indicating that **this API is convergent**, and when developers need more image formats, `please use the third-party image library`.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by eggfly** [comment]
@dnfield @tvolkert @gaaclarke please take a short review 😄 thank u.

**Review by dnfield — CHANGES_REQUESTED** [review]
I think this has been discussed before. We would rather not have to ship additional encoders.

**Comment by eggfly** [comment]
> 



> I think this has been discussed before. We would rather not have to ship additional encoders.

@dnfield  Or I just added a single line comment for developers, guiding to use third-party image library?

**Comment by eggfly** [comment]
@dnfield I change the goal of this PR, please review again~

File: lib/ui/painting.dart [hunk 1]
```diff
@@ -1574,6 +1574,10 @@ class Paint {
 
 /// The format in which image bytes should be returned when using
 /// [Image.toByteData].
+///
+/// NOTE: This ImageByteFormat enum needs to converge and will not add more
```
Comment (tvolkert) [inline lib/ui/painting.dart:1578]: Nit: `s/NOTE://`

(and below)

Comment (tvolkert) [reply lib/ui/painting.dart:1578]: I don't think it needs to converge (we're not going to reduce the values in the enum) -- just say not to add new values to the enum.

Comment (dnfield) [reply lib/ui/painting.dart:1578]: +1 - we should just make this a `//` comment, and ideally link to the github issue history for context.

Comment (eggfly) [reply lib/ui/painting.dart]: Done! Thanks

**Review by tvolkert — COMMENTED** [review]
Can you still change these to `//` comments instead of `///` comments?  Thanks!

**Comment by eggfly** [comment]
> Can you still change these to `//` comments instead of `///` comments? Thanks!

Sorry, I forgot that. Should I put my `//` comments under the `///` dartdoc?

**Review by dnfield — APPROVED** [review]
Lgtm

**Review by tvolkert — APPROVED** [review]
LGTM

**Pull request merged**